### PR TITLE
DHFPROD-3430: Update examples to work with 5.1 (develop branch)

### PR DESCRIPTION
**Pull request by ryanjdew**

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mastering/default/lib.sjs [hunk 1]
```diff
@@ -21,6 +21,7 @@ const masteringConsts = require("/com.marklogic.smart-mastering/constants.xqy");
 const requiredOptionProperties = ['matchOptions', 'mergeOptions'];
 const ps = require('/MarkLogic/provenance');
 const op = require('/MarkLogic/optic');
+const emptySequence = Sequence.from([]);
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mastering/default/lib.sjs]: I think this and the other two application code changes really need to be in a separate PR, ideally associated with a separate JIRA ticket that captures the problems these are fixing. Also, those can in theory be moved along faster than the rest of this PR - i.e. the rest of this PR doesn't seem as important as these 3 changes here, since the rest of the PR is just updates to example projects.

**Comment by ryanjdew** [comment]
@wooldridge I'm not seeing that error. Below is what I'm getting when running the step. Can you perhaps share the output of `./gradlew --version -Ptesting`?

```./gradlew hubRunFlow -PflowName=ingestion_only-flow -Ptesting

> Configure project :
input directory is extracted to: /Users/rdew/space/marklogic/marklogic-data-hub/examples/dh-5-example/input

> Task :hubRunFlow
Running Flow: [ingestion_only-flow]


Output:
{
    "user": "admin",
    "jobId": "3d02ed11-c083-467a-a3e7-d101d474f340",
    "stepResponses": {
        "1": {
            "stepOutput": null,
            "failedBatches": 0,
            "stepDefinitionName": "default-ingestion",
            "failedEvents": 0,
            "jobId": "3d02ed11-c083-467a-a3e7-d101d474f340",
            "status": "completed step 1",
            "successfulEvents": 829,
            "stepName": "json-ingest-step-json",
            "flowName": "ingestion_only-flow",
            "fullOutput": null,
            "stepEndTime": "2019-11-21T17:19:33.97146Z",
            "successfulBatches": 9,
            "stepStartTime": "2019-11-21T17:19:28.37894Z",
            "totalEvents": 829,
            "success": true,
            "stepDefinitionType": "ingestion"
        },
        "2": {
            "stepOutput": null,
            "failedBatches": 0,
            "stepDefinitionName": "default-ingestion",
            "failedEvents": 0,
            "jobId": "3d02ed11-c083-467a-a3e7-d101d474f340",
            "status": "completed step 2",
            "successfulEvents": 829,
            "stepName": "xml-ingest-step-xml",
            "flowName": "ingestion_only-flow",
            "fullOutput": null,
            "stepEndTime": "2019-11-21T17:19:37.387529Z",
            "successfulBatches": 9,
            "stepStartTime": "2019-11-21T17:19:34.013126Z",
            "totalEvents": 829,
            "success": true,
            "stepDefinitionType": "ingestion"
        },
        "3": {
            "stepOutput": null,
            "failedBatches": 0,
            "stepDefinitionName": "default-ingestion",
            "failedEvents": 0,
            "jobId": "3d02ed11-c083-467a-a3e7-d101d474f340",
            "status": "completed step 3",
            "successfulEvents": 1261,
            "stepName": "csv-ingest-step-json",
            "flowName": "ingestion_only-flow",
            "fullOutput": null,
            "stepEndTime": "2019-11-21T17:19:43.129381Z",
            "successfulBatches": 13,
            "stepStartTime": "2019-11-21T17:19:37.421629Z",
            "totalEvents": 1261,
            "success": true,
            "stepDefinitionType": "ingestion"
        },
        "4": {
            "stepOutput": null,
            "failedBatches": 0,
            "stepDefinitionName": "default-ingestion",
            "failedEvents": 0,
            "jobId": "3d02ed11-c083-467a-a3e7-d101d474f340",
            "status": "completed step 4",
            "successfulEvents": 1261,
            "stepName": "csv-ingest-step-xml",
            "flowName": "ingestion_only-flow",
            "fullOutput": null,
            "stepEndTime": "2019-11-21T17:19:47.803889Z",
            "successfulBatches": 13,
            "stepStartTime": "2019-11-21T17:19:43.162539Z",
            "totalEvents": 1261,
            "success": true,
            "stepDefinitionType": "ingestion"
        },
        "5": {
            "stepOutput": null,
            "failedBatches": 0,
            "stepDefinitionName": "default-ingestion",
            "failedEvents": 0,
            "jobId": "3d02ed11-c083-467a-a3e7-d101d474f340",
            "status": "completed step 5",
            "successfulEvents": 1,
            "stepName": "ingest-binary",
            "flowName": "ingestion_only-flow",
            "fullOutput": null,
            "stepEndTime": "2019-11-21T17:19:47.932907Z",
            "successfulBatches": 1,
            "stepStartTime": "2019-11-21T17:19:47.833494Z",
            "totalEvents": 1,
            "success": true,
            "stepDefinitionType": "ingestion"
        },
        "6": {
            "stepOutput": null,
            "failedBatches": 0,
            "stepDefinitionName": "default-ingestion",
            "failedEvents": 0,
            "jobId": "3d02ed11-c083-467a-a3e7-d101d474f340",
            "status": "completed step 6",
            "successfulEvents": 829,
            "stepName": "json-ingest-step-text",
            "flowName": "ingestion_only-flow",
            "fullOutput": null,
            "stepEndTime": "2019-11-21T17:19:50.058117Z",
            "successfulBatches": 9,
            "stepStartTime": "2019-11-21T17:19:47.971306Z",
            "totalEvents": 829,
            "success": true,
            "stepDefinitionType": "ingestion"
        },
        "7": {
            "stepOutput": null,
            "failedBatches": 0,
            "stepDefinitionName": "default-ingestion",
            "failedEvents": 0,
            "jobId": "3d02ed11-c083-467a-a3e7-d101d474f340",
            "status": "completed step 7",
            "successfulEvents": 829,
            "stepName": "xml-ingest-step-text",
            "flowName": "ingestion_only-flow",
            "fullOutput": null,
            "stepEndTime": "2019-11-21T17:19:54.24353Z",
            "successfulBatches": 9,
            "stepStartTime": "2019-11-21T17:19:50.104979Z",
            "totalEvents": 829,
            "success": true,
            "stepDefinitionType": "ingestion"
        }
    },
    "flowName": "ingestion_only-flow",
    "endTime": "2019-11-21T17:19:54.347489Z",
    "lastCompletedStep": "7",
    "lastAttemptedStep": "7",
    "jobStatus": "finished",
    "startTime": "2019-11-21T17:19:28.337814Z"
}

BUILD SUCCESSFUL in 30s
1 actionable task: 1 executed
```
